lib/elasticsearch_models/base.rb

```diff
@@ -22,17 +22,26 @@ class CreateError < StandardError; end
 
     class << self
       def create!(*params)
+        model    = build!(*params)
+        response = insert_squashed_model_into_store(model.deep_squash_to_store, model.index_name)
+        model.assign_metadata_fields(response)
+        model
+      end
+
+      def build!(*params)
         model = new(*params)
         model.rehydration_class = type
         model.query_types       = query_types
         model.validate!
-        response = client_connection.index(index: model.index_name, type: DEPRECATED_TYPE, body: model.deep_squash_to_store)
+        model
+      end
 
+      def insert_squashed_model_into_store(squashed_model, index)
```

Comment:
A few suggestions to help the method read a bit better and clarify the method's responsibilities
* `insert!` reads a bit better
* `squashed_model` could be renamed to `body_hash` to indicate a hash is expected and not a model
* `body_hash.is_a?(Hash) or raise` can ensure it's a hash and prevent exceptions from raising from the `elasticsearch` gem (which may be confusing)